conwayste/src/uimanager.rs

```diff
@@ -0,0 +1,140 @@
+/*  Copyright 2019 the Conwayste Developers.
+ *
+ *  This file is part of conwayste.
+ *
+ *  conwayste is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  conwayste is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with conwayste.  If not, see
+ *  <http://www.gnu.org/licenses/>. */
+
+use std::collections::HashMap;
+use chromatica::css;
+
+use ggez::graphics::{self, Rect, Font, Color, DrawMode, DrawParam, Text, Scale};
+use ggez::nalgebra::{Point2, Vector2};
+use ggez::{Context, GameResult};
+
+use crate::config::Config;
+use crate::ui::{
+    Widget,
+    Button,
+    Checkbox, ToggleState,
+    Chatbox,
+    Layer,
+    Pane,
+    Screen,
+    TextField, TextInputState,
+    UIAction,
+    WidgetID,
+};
+
+pub struct UIManager {
+    layers: HashMap<Screen, Vec<Layer>>,
+}
+
+impl UIManager {
+    pub fn new(ctx: &mut Context, config: &Config) -> Self {
+        let font = graphics::Font::default(); // Provides DejaVuSerif.ttf
+        let mut ui_layers = HashMap::new();
+
+        let chat_pane_rect = Rect::new(30.0, 600.0, 300.0, 150.0);
```

Comment:
I feel like this isn't the right place for this code. The UIManager shouldn't contain code for any particular UI layout -- instead the user of it would do all that.